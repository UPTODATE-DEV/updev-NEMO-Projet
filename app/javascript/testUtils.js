/* eslint-disable import/no-extraneous-dependencies */
import { mount } from 'enzyme';

/**
 * Stubbed Bootstrap components seem to cause these warnings
 * when `mount`ed with Enzyme (though react-test-renderer is fine).
 */
export const STUB_COMPONENT_WARNINGS = /Warning: (<.*> is using incorrect casing|The tag <.*> is unrecognized in this browser|React does not recognize the `.*` prop on a DOM element|Unknown event handler property `.*`. It will be ignored.|Received `.*` for a non-boolean attribute `.*`.)/;

// eslint-disable-next-line no-console
const originalConsoleError = console.error;

export function suppressErrors(matcher = /^$/) {
  // eslint-disable-next-line no-console
  console.error = (...messages) => {
    if (messages[0].match(matcher)) {
      return;
    }

    originalConsoleError(...messages);
  };
}

export function unsuppressAllErrors() {
  // eslint-disable-next-line no-console
  console.error = originalConsoleError;
}

export function quietMount(component) {
  suppressErrors(STUB_COMPONENT_WARNINGS);
  const wrapper = mount(component);
  unsuppressAllErrors();

  return wrapper;
}
