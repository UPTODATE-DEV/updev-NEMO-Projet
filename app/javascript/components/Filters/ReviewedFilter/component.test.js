import React from 'react';
import { shallow } from 'enzyme';

import { getFiltersStore } from '../testUtils';

import Component from './component';
import { quietMount } from '../../../testUtils';

const defaultProps = {
  filtersStore: getFiltersStore(),
  onSubmit: jest.fn(),
};

it('renders as expected', () => {
  const wrapper = shallow(<Component {...defaultProps} />);
  expect(wrapper).toMatchSnapshot();
});

describe('popover', () => {
  const wrapper = quietMount(<Component {...defaultProps} />);
  wrapper.find('Button#reviewed-filter').simulate('click');

  const overlay = quietMount(wrapper.find('OverlayTrigger').prop('overlay'));

  it('handles callbacks', () => {
    overlay.find('#no').simulate('click');
    overlay.find('Button.btn-apply').simulate('click');

    expect(defaultProps.onSubmit.mock.calls.length).toMatchSnapshot();
    expect(defaultProps.onSubmit.mock.results).toMatchSnapshot();
  });
});
