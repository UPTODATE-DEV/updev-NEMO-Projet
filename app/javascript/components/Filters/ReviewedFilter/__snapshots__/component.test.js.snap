// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`popover handles callbacks 1`] = `1`;

exports[`popover handles callbacks 2`] = `
Array [
  Object {
    "type": "return",
    "value": undefined,
  },
]
`;

exports[`renders as expected 1`] = `
<FilterOverlayTrigger
  buttonClass="btn-margin-left"
  hints={null}
  id="reviewed-filter"
  onSubmit={[MockFunction]}
  popoverContent={
    <React.Fragment>
      <div>
        <Label>
          Is marked 'reviewed'
        </Label>
      </div>
      <ButtonGroup>
        <Button
          active={true}
          id="yes"
          onClick={[Function]}
          variant="secondary"
        >
          Yes
        </Button>
        <Button
          active={false}
          id="no"
          onClick={[Function]}
          variant="secondary"
        >
          No
        </Button>
        <Button
          active={false}
          id="either"
          onClick={[Function]}
          variant="secondary"
        >
          Either
        </Button>
      </ButtonGroup>
    </React.Fragment>
  }
  title="Reviewed"
/>
`;
